skimage/_shared/utils.py

```diff
@@ -20,6 +20,44 @@ class skimage_deprecation(Warning):
     pass
 
 
+class deprecate_arg:
+    """Decorator ensuring backward compatibility when argument names are
+    modified in a function definition.
+
+    Parameters
+    ----------
+    arg_mapping: dict
+        mapping between the function's old argument names and the new ones.
+    removed_version : str
+        The package version in which the deprecated argument will be removed.
+
+    """
+
+    def __init__(self, arg_mapping, removed_version=None):
+        self.arg_mapping = arg_mapping
+        self.warning_msg = "'%s' is a deprecated argument name for %s. "
+        if removed_version is not None:
+            self.warning_msg += "It will be removed in version %s" % (
+                removed_version)
+        self.warning_msg += "Use '%s' instead."
+
+    def __call__(self, func):
```

Comment:
I think we would want tests for this.

Reply:
I did not implement tests because I found that the `deprecated` decorator is not tested neither, but I can manage to do it, the way it is made in `wabisabi` ;-)

Reply:
Are the implemented tests good enough?